File: CodeGeneratorMVC/oldClasses/ConnectionString.vb
Option Strict On
<Serializable()> Public Class ConnectionString
    Private _ID As Integer
    Private _Name As String
    Private _UserName As String
    Public Sub New()

    End Sub
    Public Sub New(ByVal ConnectionID As Integer, ByVal ConnectionName As String, ByVal ConnectionUserName As String)
        _ID = ConnectionID
        _Name = ConnectionName
        _UserName = ConnectionUserName
    End Sub
    Public Property ID() As Integer
        Get
            Return _ID
        End Get
        Set(ByVal value As Integer)
            _ID = value
        End Set
    End Property
    Public Property Name() As String
        Get
            Return _Name
        End Get
        Set(ByVal value As String)
            _Name = value
        End Set
    End Property
    Public Property UserName() As String
        Get
            Return _UserName
        End Get
        Set(ByVal value As String)
            _UserName = value
        End Set
    End Property
    Public ReadOnly Property Self() As ConnectionString
        Get
            Return Me
        End Get
    End Property
    Public Overrides Function ToString() As String
        Return _Name
    End Function
    Public Overrides Function Equals(ByVal obj As Object) As Boolean
        If TypeOf obj Is ConnectionString Then
            Return Me = CType(obj, ConnectionString)
        End If
        Return False
    End Function
    Public Overloads Shared Operator =(ByVal pv1 As ConnectionString, ByVal pv2 As ConnectionString) As Boolean
        If pv1 Is Nothing And pv2 Is Nothing Then
            Return True
        End If
        If pv1 Is Nothing OrElse pv2 Is Nothing Then
            Return False
        End If
        Return pv1.ID = pv2.ID
    End Operator
    Public Overloads Shared Operator <>(ByVal pv1 As ConnectionString, ByVal pv2 As ConnectionString) As Boolean
        Return Not pv1 = pv2
    End Operator




End Class

File: CodeGeneratorMVC/Models/ConnectionString.cs
using System;

[Serializable()]
public class ConnectionString {
    private int _ID;
    private string _Name;
    private string _UserName;
    public ConnectionString() {
    }
    public ConnectionString(int ConnectionID, string ConnectionName, string ConnectionUserName) {
        _ID = ConnectionID;
        _Name = ConnectionName;
        _UserName = ConnectionUserName;
    }
    public int ID {
        get {
            return _ID;
        }
        set {
            _ID = value;
        }
    }
    public string Name {
        get {
            return _Name;
        }
        set {
            _Name = value;
        }
    }
    public string UserName {
        get {
            return _UserName;
        }
        set {
            _UserName = value;
        }
    }
    public ConnectionString Self {
        get {
            return this;
        }
    }
    public override string ToString() {
        return _Name;
    }
    public override bool Equals(object obj) {
        if (obj is ConnectionString)
            return this == (ConnectionString)obj;
        return false;
    }
    public static bool operator ==(ConnectionString pv1, ConnectionString pv2) {
        if (pv1 == null & pv2 == null)
            return true;
        if (pv1 == null || pv2 == null)
            return false;
        return pv1.ID == pv2.ID;
    }
    public static bool operator !=(ConnectionString pv1, ConnectionString pv2) {
        return pv1 != pv2;
    }
}
